File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode c,f1,sc;
        ListNode t1=new ListNode(-1);
        t1.next=head;
        c=t1;
        while(c.next!=null && c.next.next!=null){
            f1=c.next;
            sc=c.next.next;
            f1.next=sc.next;
            c.next=sc;
            c.next.next=f1;
            c=c.next.next;
        }
        
        return t1.next;
    }
}

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
          if(head==NULL || head->next==NULL)
          return head;
          ListNode *s=NULL,*c=head,*c1=head->next,*f=NULL;
         
          while(c!=NULL&&c->next!=NULL){
      
            f=c->next;
            s=c->next->next;
            f->next=c;
            if (s==NULL||s->next==NULL)
            c->next=s;
            else
            c->next=s->next;
            c=s;
            
          }
          return c1;                  
    }
};